File: Parking_Management_System/VehicleFactory.cpp
#include "VehicleFactory.hpp"
#include "Car.hpp"
#include "Bike.hpp"
using namespace std;

Vehicle* VehicleFactory::createVehicle(string type, string license){
    if(type == "Bike"){
        return new Bike(license);
    }else{
        return new Car(license);
    }
}

File: Java/Parking Management System/src/main/java/org/example/Vehicle/VehicleFactory.java
package org.example.Vehicle;

public class VehicleFactory {
    public Vehicle createVehicle(String type, String license, VehicleSize size){
        if(type == "Bike"){
            return new Bike(license, size);
        }else{
            return new Car(license, size);
        }
    }
}
